xray/backends/netCDF4_.py

```diff
@@ -164,6 +171,18 @@ def set_variable(self, name, variable):
             fill_value = None
 
         encoding = variable.encoding
+        endian = encoding.get('endian', _endian_lookup[datatype.byteorder])
```

Comment:
`datatype` here can be `str` (as set on line 160 above). This is why we're getting a build failure on Python 3 with this change.